src/dodal/beamlines/i22.py

```diff
@@ -41,338 +39,233 @@
-) -> TetrammDetector:
-    return device_instantiation(
-        TetrammDetector,
-        "it",
-        "-EA-TTRM-02:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def it():
+    """Create an IT detector with specific settings."""
+    return TetrammDetector(
+        name="it",
+        prefix="-EA-TTRM-02:",
         type="PIN Diode",
         directory_provider=get_directory_provider(),
     )
 
 
-def vfm(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> FocusingMirror:
-    return device_instantiation(
-        FocusingMirror,
-        "vfm",
-        "-OP-KBM-01:VFM:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def vfm():
+    """Create a VFM instance with specific settings."""
+    return FocusingMirror(name="vfm", prefix="-OP-KBM-01:VFM:")
 
 
-def hfm(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> FocusingMirror:
-    return device_instantiation(
-        FocusingMirror,
-        "hfm",
-        "-OP-KBM-01:HFM:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def hfm():
+    """Create an HFM instance with specific settings."""
+    return FocusingMirror(name="hfm", prefix="-OP-KBM-01:HFM:")
 
 
-def dcm(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> DoubleCrystalMonochromator:
-    return device_instantiation(
-        DoubleCrystalMonochromator,
-        "dcm",
-        "",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-        bl_prefix=False,
-        motion_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
-        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
-        crystal_1_metadata=CrystalMetadata(
-            usage="Bragg",
-            type="silicon",
-            reflection=(1, 1, 1),
-            d_spacing=(3.13475, "nm"),
-        ),
-        crystal_2_metadata=CrystalMetadata(
-            usage="Bragg",
-            type="silicon",
-            reflection=(1, 1, 1),
-            d_spacing=(3.13475, "nm"),
-        ),
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def dcm():
+    """Create a DCM instance with specific settings."""
+    prefix = BeamlinePrefix(BL).beamline_prefix
+    silicon_crystal_metadata = CrystalMetadata(
+        usage="Bragg",
+        type="silicon",
+        reflection=(1, 1, 1),
+        d_spacing=(3.13475, "nm"),
     )
 
+    return DoubleCrystalMonochromator(
+        name="dcm",
+        prefix="",
+        motion_prefix=f"{prefix}-MO-DCM-01:",
+        temperature_prefix=f"{prefix}-DI-DCM-01:",
+        crystal_1_metadata=silicon_crystal_metadata,
```

Comment:
I am not sure of the spacing, as an i22 matter this needs to wait until the PBS is back I think; regardless this is not in the PR scope

Reply:
This isn't about whether the values are correct, it's whether dcm.read() returns the signals twice or whether some dict/set uniqueness handling will prevent both subdevices being read simultaneously: since this PR is moving from having 2 identical devices to having 1 device that is intended to be read twice, it's relevant to the PR scope.